test/Gradient/TestTypeConversion.C

```diff
@@ -0,0 +1,59 @@
+// RUN: %cladnumdiffclang -lm -lstdc++ %s  -I%S/../../include -oGradients.out 2>&1 | FileCheck %s
+// RUN: ./Gradients.out | FileCheck -check-prefix=CHECK-EXEC %s
+
+//CHECK-NOT: {{.*error|warning|note:.*}}
+
+#include "clad/Differentiator/Differentiator.h"
+#include <cmath>
+
+#include "../TestUtils.h"
+
+float fn_type_conversion(float z, int a) { 
+  for (int i = 1; i < a; i++){
+    z = z * a;
+  }
+  return z;
+}
+
+void fn_type_conversion_grad(float z, int a, clad::array_ref<float> _d_z, clad::array_ref<int> _d_a);
+// CHECK: void fn_type_conversion_grad(float z, int a, clad::array_ref<float> _d_z, clad::array_ref<int> _d_a) {
+// CHECK-NEXT:     unsigned long _t0;
+// CHECK-NEXT:     int _d_i = 0;
+// CHECK-NEXT:     clad::tape<int> _t1 = {};
+// CHECK-NEXT:     clad::tape<float> _t2 = {};
+// CHECK-NEXT:     _t0 = 0;
+// CHECK-NEXT:     for (int i = 1; i < a; i++) {
+// CHECK-NEXT:         _t0++;
+// CHECK-NEXT:         z = clad::push(_t2, z) * clad::push(_t1, a);
+// CHECK-NEXT:     }
+// CHECK-NEXT:     float fn_type_conversion_return = z;
+// CHECK-NEXT:     goto _label0;
+// CHECK-NEXT:   _label0:
+// CHECK-NEXT:     * _d_z += 1;
+// CHECK-NEXT:     for (; _t0; _t0--) {
+// CHECK-NEXT:         {
+// CHECK-NEXT:             float _r_d0 = * _d_z;
+// CHECK-NEXT:             float _r0 = _r_d0 * clad::pop(_t1);
+// CHECK-NEXT:             * _d_z += _r0;
+// CHECK-NEXT:             float _r1 = clad::pop(_t2) * _r_d0;
+// CHECK-NEXT:             * _d_a += _r1;
+// CHECK-NEXT:             * _d_z -= _r_d0;
+// CHECK-NEXT:             * _d_z;
+// CHECK-NEXT:         }
+// CHECK-NEXT:     }
+// CHECK-NEXT: }
+
+#define TEST(F, x, y)                                                          \
+  {                                                                            \
+    result_0 = 0;                                                             \
+    result_1 = 0;                                                             \
+    clad::gradient(F);                                                         \
+    F##_grad(x, y, &result_0, &result_1);                                    \
+    printf("Result is = {%.2f, %.2f}\n", result_0, (float)result_1);                \
+  }
+
+int main() {
+  float result_0;
+  int result_1;
+  TEST(fn_type_conversion, 4, 3); // CHECK-EXEC: Result is = {9.00, 24.00}
```

Comment:
```suggestion
  TEST_GRADIENT(fn_type_conversion, /*numOfDerivativeArgs=*/2, 4, 3, &result_0, &result_1); // CHECK-EXEC: Result is = {9.00, 24.00}
```

`TEST_GRADIENT` is defined in `TestUtils.h`.

Reply:
> 



> `TEST_GRADIENT` is defined in `TestUtils.h`.

I tried but it shows some error message like:
```
/home/phrygiangates/gsoc/clad/test/Gradient/../TestUtils.h:92:5: error: member reference base type 'void (*)(float, int, clad::array_ref<float>, clad::array_ref<int>)' is not a structure or union
  cf.execute(args...);
```

Reply:
Yes.

Change:
```cpp
TEST(fn_type_conversion, 4, 3);
```
this to:

```cpp
INIT_GRADIENT(fn_type_conversion);
TEST_GRADIENT(fn_type_conversion, /*numOfDerivativeArgs=*/2, 4, 3, &result_0, &result_1); // CHECK-EXEC: {9.00, 24}
```

And also, add an overload `void print(float f)` in `TestUtils.h`. You just have to add the below code in TestUtils.h

```cpp
void print(float f) { printf("%.2f", f); }
```

Reply:
Thank you so much!